myapp/app/controllers/tasks_controller.rb

```diff
@@ -41,6 +41,15 @@ def destroy
     end
   end
 
+  def search
+    if params[:name].nil? && params[:status].nil?
+      @tasks = Task.all
+    else
+      @tasks = Task.name_like(params[:name]).status_equal(Task.statuses[params[:status]])
+    end
```

Comment:
こちら確認してみてください。
```suggestion
    @tasks = Task.name_like(params[:name]).status_equal(Task.statuses[params[:status]])
```
おそらくscopeでpresent?チェックを行なっているため、45行目のnilをチェックせずとも同様の動きになるかと思います。
params[:name]もparams[:status]が空文字だとしても結果「Task.all」と同じ挙動になるはずですので確認お願いします。

Reply:
おっしゃる通り、分岐処理は不要でしたので削除しました。